Scripts/WikiUpdater.cs

```diff
@@ -0,0 +1,415 @@
+            translationPairs.Add(untranslatedPage.Name, translatedPage.Name);
+
+            for (var j = 0; j < untranslatedPage.Sections.Count; j++)
+            {
+                var untranslatedSection = untranslatedPage.Sections[j];
+                var translatedSection = translatedPage.Sections[j];
+
+                // Translate body sections
+                translationPairs.Add(untranslatedSection.SectionBody, translatedSection.SectionBody);
+
+                if (untranslatedSection.SectionHeading != null && translatedSection.SectionHeading != null)
+                {
+                    // Translate headings if present and not already in the list to translate
+                    translationPairs.TryAdd(untranslatedSection.SectionHeading, translatedSection.SectionHeading);
+                }
+            }
+        }
+
+        var reader = File.ReadLinesAsync(ENGLISH_TRANSLATION_FILE, Encoding.UTF8, cancellationToken);
+        var writer = new StreamWriter(File.Create(TEMP_TRANSLATION_FILE), new UTF8Encoding(false));
+
+        var isReplacingValue = false;
+        var isFuzzyValue = false;
+        await foreach (var line in reader)
+        {
+            if (line == string.Empty)
+            {
+                // Blank lines mark the end of a value we might be replacing, so stop replacing
+                isReplacingValue = false;
+                await writer.WriteLineAsync(line);
+                continue;
+            }
+
+            if (isReplacingValue)
+            {
+                // Skip lines for values we're replacing
+                continue;
+            }
+
+            if (line == "#, fuzzy")
+            {
+                // Defer adding fuzzy labels so we can include/exclude them based on the next line
+                isFuzzyValue = true;
+                continue;
+            }
+
+            if (!line.Contains("msgid \""))
+            {
+                // Copy existing lines that we don't want to replace
+                await writer.WriteLineAsync(line);
+                continue;
+            }
+
+            var key = line.Split("\"")[1];
+            if (translationPairs.TryGetValue(key, out var value))
+            {
+                // Skip fuzzy labels if present, since we know this value
+                isFuzzyValue = false;
+                isReplacingValue = true;
+
+                // Add the key line
+                await writer.WriteLineAsync(line);
+
+                // Add the value line(s)
+                var linesToInsert = value.Split("\n");
+                if (linesToInsert.Length == 1)
+                {
+                    await writer.WriteLineAsync($"msgstr \"{value}\"");
+                }
+                else
+                {
+                    await writer.WriteLineAsync("msgstr \"\"");
+
+                    // Split the content over multiple lines with correct formatting
+                    for (var j = 0; j < linesToInsert.Length; j++)
```

Comment:
I'm seeing a j here as well. Also preincrement  (`++i`) should be used instead of postincrement in loops.